horovod/spark/common/store.py

```diff
@@ -0,0 +1,293 @@
+# Copyright 2019 Uber Technologies, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import contextlib
+import os
+import shutil
+import tempfile
+
+import pyarrow as pa
+
+
+HDFS_PREFIX = 'hdfs://'
+
+
+class Store(object):
+    def get_filesystem(self):
+        raise NotImplementedError()
+
+    def get_train_data_path(self):
+        raise NotImplementedError()
+
+    def get_val_data_path(self):
+        raise NotImplementedError()
+
+    def get_test_data_path(self):
+        raise NotImplementedError()
+
+    def saving_runs(self):
+        raise NotImplementedError()
+
+    def get_runs_path(self):
+        raise NotImplementedError()
+
+    def get_run_path(self, run_id):
+        raise NotImplementedError()
+
+    def get_checkpoint_path(self, run_id):
+        raise NotImplementedError()
+
+    def get_logs_path(self, run_id):
+        raise NotImplementedError()
+
+    def get_checkpoint_filename(self):
+        raise NotImplementedError()
+
+    def get_logs_subdir(self):
+        raise NotImplementedError()
+
+    def exists(self, path):
```

Comment:
In its current form you're definitely right, but I want to keep Store generic enough that it could potentially support storage APIs other than filesystems (like some kind of blob storage DB).  Does that seem reasonable to you?